restcontrollers/projectController.go

```diff
@@ -108,195 +102,166 @@ func (c *RESTController) addProject(w http.ResponseWriter, r *http.Request) {
 	if err := checkRequestType(GET, w, r); err != nil {
+		writeError(err.Error(), w, http.StatusBadRequest)
 		return
 	}
 
 	ids, ok := r.URL.Query()["product_id"]
 	if !ok || len(ids[0]) < 1 {
-		w.WriteHeader(http.StatusBadRequest)
-		fmt.Fprint(w, errors.New("Url Param 'product_id' is missing"))
+		writeError("Url Param 'product_id' is missing", w, http.StatusBadRequest)
 		return
 	}
 
 	id, err := uuid.Parse(ids[0])
 	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		fmt.Fprint(w, err)
+		writeError(err.Error(), w, http.StatusBadRequest)
 	}
 
 	projectList, err := c.DBController.GetProjectsByProductID(&id)
 	if err == nil {
 		b, err := json.Marshal(projectList)
 		if err != nil {
-			err = errors.Wrap(errors.WithStack(err), "Failed to encode response")
-			w.WriteHeader(http.StatusInternalServerError)
-			fmt.Fprint(w, err.Error())
+			writeError(err.Error(), w, http.StatusInternalServerError)
 			return
 		}
 
-		w.WriteHeader(http.StatusOK)
-		fmt.Fprint(w, string(b))
+		writeData(string(b), w, http.StatusOK)
 		return
 	}
 
 	if err.Error() == dbcontrollers.ErrNoProjectForProduct.Error() {
-		w.WriteHeader(http.StatusAccepted)
-		fmt.Fprint(w, err.Error())
+		writeError(err.Error(), w, http.StatusAccepted)
 		return
 	}
-	err = errors.Wrap(errors.WithStack(err), "Failed to get product projects")
-	w.WriteHeader(http.StatusInternalServerError)
-	fmt.Fprint(w, err.Error())
+	writeError(err.Error(), w, http.StatusInternalServerError)
 }
 
 func (c *RESTController) getProjects(w http.ResponseWriter, r *http.Request) {
 	log.Println("Getting multiple projects")
 	if err := checkRequestType(GET, w, r); err != nil {
+		writeError(err.Error(), w, http.StatusBadRequest)
 		return
 	}
 
 	idList, err := parseIDList(w, r)
 	if err != nil {
-		fmt.Fprint(w, err)
+		writeError(err.Error(), w, http.StatusBadRequest)
 		return
 	}
 
 	projectData, err := c.DBController.GetProjects(idList)
 	if err == nil {
 		b, err := json.Marshal(projectData)
 		if err != nil {
-			err = errors.Wrap(errors.WithStack(err), "Failed to encode response")
-			w.WriteHeader(http.StatusInternalServerError)
-			fmt.Fprint(w, err.Error())
+			writeError(err.Error(), w, http.StatusInternalServerError)
 			return
 		}
 
-		w.WriteHeader(http.StatusOK)
-		fmt.Fprint(w, string(b))
+		writeData(string(b), w, http.StatusOK)
 		return
 	}
 
 	if err.Error() == dbcontrollers.ErrProjectNotFound.Error() {
-		w.WriteHeader(http.StatusAccepted)
-		fmt.Fprint(w, err.Error())
+		writeError(err.Error(), w, http.StatusAccepted)
 		return
 	}
 	err = errors.Wrap(errors.WithStack(err), "Failed to get projects")
-	w.WriteHeader(http.StatusInternalServerError)
-	fmt.Fprint(w, err.Error())
+	writeError(err.Error(), w, http.StatusInternalServerError)
 }
 
 func (c *RESTController) deleteProject(w http.ResponseWriter, r *http.Request) {
 	log.Println("Deleting project")
 	data, err := decodePostData(w, r)
 	if err != nil {
+		writeError(err.Error(), w, http.StatusBadRequest)
```

Comment:
If I were you I think I create a new wrapper around http.ResponseWriter and bind writeError to that new item/struct/interface.
`func (w ResponseWriter) writeError(err error, status int) `

and use it in the following way:
`w.writeError(err.Error(), http.StatusBadRequest)`